File: BinaryTree/python/treeValueCount.py
from typing import Optional

from node import Node


def tree_value_count(root: Optional[Node], target: int) -> int:
    if root is None:
        return 0
    match = 1 if root.val == target else 0
    return match + tree_value_count(root.left, target) + tree_value_count(root.right, target)

File: BinaryTree/go/treeValueCount.go
package binarytree

// TreeValueCount counts how many nodes equal target.
func TreeValueCount(root *TreeNode, target int) int {
	if root == nil {
		return 0
	}
	match := 0
	if root.Val == target {
		match = 1
	}
	return match + TreeValueCount(root.Left, target) + TreeValueCount(root.Right, target)
}
